Frontend development

baloveless opened the pull request:
The team has completed the user information and change user information page, and there is a skeleton of the redirect page which will be completed on the next merge. Some tests need to be fixed, I won't merge the PR until those are sorted. This merge will leave only a handful of changes until MVP, the largest being the change password page. I also added some changes to the layout.

Jim961128 committed:
Updated Profile page

Jim961128 committed:
Updated User info page

Jim961128 committed:
trying to add .cs inside .gitignore

Jim961128 committed:
Added .vs/

Jim961128 committed:
trying to fix the merge

baloveless committed:
fleshed out profile page and fixed links
to increase accessibility

baloveless committed:
Finished basis of profile page
Also readded redirect cause I think it was lost in a merge

baloveless committed:
fixed redirect page.
Need to make changes to profile page edit before MVP.
Might ask for help on that cause it isn't too crazy.

baloveless committed:
Profile page allows for editting and requires user
to be signed in. editing will not activate if they are not signed in.

baloveless committed:
Added some redirects for whether a user is logged.

baloveless committed:
fixed a bad line

baloveless committed:
Set up for team meeting afternoon 4/16

baloveless committed:
Merge branch 'Work' into Frontend-development

baloveless committed:
Testing demonstration with the boys

baloveless committed:
before trying to merge in backend changes

baloveless committed:
Merge branch 'Work' into Frontend-development

baloveless committed:
Merge branch 'development' into Frontend-development

baloveless committed:
getting front and back working locally

baloveless committed:
tried to fix dependency issues

baloveless committed:
forgot a package

baloveless committed:
Need to iron out why my screen shows errors
on css module imports, but it works when ran.
So it's fine other than being an unsightly annoyance.

Jim961128 committed:
added profile test

Jim961128 committed:
Merge branch 'Frontend-development' of https://github.com/PSU-CapstoneBrownHope/BrownHopeAidApp into Frontend-development

baloveless committed:
did some css work on profile page

Jim961128 committed:
added textbox test for Profile page

adrian27-pdx committed:
SignUp Unit Test

baloveless committed:
css tweeks

baloveless committed:
Merge branch 'Frontend-development' of https://github.com/PSU-CapstoneBrownHope/BrownHopeAidApp into Frontend-development

baloveless committed:
Some merge issues on Profile.test.tsx

baloveless committed:
fixing user check

baloveless committed:
Update Account info isn't working but
it does display user info so that is a page done.

baloveless committed:
Need to figure out infinite test issue in profile

baloveless committed:
Found a good formula for formatting phone
number inputs.
Added a footer for elements like a link to our github
to prepare for backlog goals.

baloveless committed:
Made the header static and each page is
now a div that can be scrolled to accommodate user information form

mackfoggia reviewed (dismissed):
It looks like we're still picking up some .vs files, I would like to keep those out of the development branch if at all possible

Jim961128 committed:
merge errors

baloveless committed:
I tweaked some small backend things

baloveless committed:
Merge branch 'Frontend-development' of https://github.com/PSU-CapstoneBrownHope/BrownHopeAidApp into Frontend-development

baloveless committed:
delete vs

baloveless commented:
> It looks like we're still picking up some .vs files, I would like to keep those out of the development branch if at all possible

I added a line to the .gitignore, hopefully .vs folder wont be there in future

baloveless committed:
fixed profile test

baloveless committed:
Merge branch 'Work' into Frontend-development

baloveless committed:
Fixed test problems for merge

mackfoggia commented on back/src/routes/airtable.ts:
```diff
@@ -105,6 +105,7 @@ airtableRouter.post('/signout', function(req, res, next) {
 
 
 airtableRouter.post("/getInfo", function (req, res) {
+  console.log(JSON.stringify(req.body[0]))
```
should this be in the pr?

mackfoggia replied:
```suggestion
```

baloveless replied:
removed that line, that was to sort out some of my confusion

mackfoggia commented on frontend/src/components/Redirect.tsx:
```diff
@@ -0,0 +1,37 @@
+  // need to find right name for these variables. 
+  const infoURL = "https://www.blackresiliencefund.com/faq"
+  const contactEmail = ""
+  const contactURL = "https://www.brownhope.org/contact"
```
The appropriate contact address for this would be info@blackresiliencefund.com

mackfoggia replied:
Just realized that's an URL and not an email, but yeah Tracy would like this email to be present for questions about the application on this page and the pages that show status.

mackfoggia commented on frontend/src/util/config.ts:
```diff
@@ -4,8 +4,8 @@ import dotenv from "dotenv"
-//const server = "localhost:5000";
-const server = '34.82.220.171';
+const server = "localhost:5000";
+//const server = '34.82.220.171';
```
I know it's a hassle but we need this to stay on the IP and not localhost for merge into development

mackfoggia replied:
marking as needs work just in case

baloveless committed:
thought this was implemented alreadu=y

baloveless committed:
fixed test, thought i already did this

baloveless committed:
Merge branch 'Work' into Frontend-development

lazyhacker7 committed:
updated readme

jjernstrom committed:
Merge pull request #140 from PSU-CapstoneBrownHope/edit-readme

updated readme

baloveless committed:
Merge branch 'Frontend-development' of https://github.com/PSU-CapstoneBrownHope/BrownHopeAidApp into Frontend-development

baloveless committed:
I made a oopsie on this one

baloveless commented:
I'm having an issue with the account information page. The route is returning status code 500. The same thing happens when I try the same with the old frontend

baloveless committed:
Austin and I talked through some changes and
all current pages work

baloveless commented:
> I'm having an issue with the account information page. The route is returning status code 500. The same thing happens when I try the same with the old frontend

That problem got resolved. This is ready to go I believe.

baloveless committed:
This completes everything but the redirect,
and application status page.
Addtions
- Username replaces profile if user is logged in
- Most pages have a feature to redirect the user when not logged in.
- Change password page works
- Changed password input fields to hide input
-- Might want to add a peak feature before final deployment
- Current page is now scrollable. scroll bar is hidden

baloveless committed:
Signup-Redirect applicaiton id passing is set up.
I tweaked the nav to check for a change in username more frequently.
Fixed accessibility issues on some pages.
Short of tests, this is at MVP in my opinion.

baloveless commented:
Read last two commits for more details, but I think this is at MVP minus a few tests.

mackfoggia commented:
If this is ready for merge, change the server to the IP instead of localhost and I will approve changes, everything looks great to me!

baloveless committed:
Fixed the server route
